packages/itwinui-react/src/core/Tooltip/Tooltip.tsx

```diff
@@ -4,58 +4,192 @@
-import { Popover, Box } from '../utils/index.js';
-import type { CommonProps, PopoverProps } from '../utils/index.js';
+import {
+  useMergeRefs,
+  useFloating,
+  autoUpdate,
+  offset,
+  flip,
+  shift,
+  useHover,
+  useFocus,
+  useDismiss,
+  useRole,
+  useInteractions,
+  useClick,
+  FloatingPortal,
+} from '@floating-ui/react';
+import type { Placement } from '@floating-ui/react';
+import { Box, type PolymorphicForwardRefComponent } from '../utils/index.js';
+import { ThemeContext } from '../ThemeProvider/ThemeContext.js';
+import ReactDOM from 'react-dom';
 
-export type TooltipProps = {
+type TooltipOptions = {
+  /**
+   * Placement of the Tooltip
+   * @default 'top'
+   */
+  placement?: Placement;
+  /**
+   * Property for manual visibility control
+   */
+  visible?: boolean;
+  /**
+   * Function that allows users to control Tooltip visibility
+   */
+  toggleVisible?: (open: boolean) => void;
+  /**
+   * Use this if you want Tooltip to follow moving trigger element
+   * @default false;
+   */
+  followTrigger?: boolean;
+};
+
+type TooltipOwnProps = {
   /**
    * Content of the tooltip.
    */
-  content: React.ReactNode;
+  content?: React.ReactNode;
   /**
    * Element to have tooltip on. Has to be a valid JSX element and needs to forward its ref.
    * If not specified, the `reference` prop should be used instead.
    */
-  children?: JSX.Element;
-} & Omit<PopoverProps, 'className'> &
-  Omit<CommonProps, 'title'>;
+  children?: React.ReactNode;
+  /**
+   * Element to append tooltip to.
+   * Appends to ThemeProvider portalContainerRef by default.
+   */
+  appendTo?: HTMLElement;
+};
 
-/**
- * Basic tooltip component to display informative content when an element is hovered or focused.
- * Uses the {@link Popover} component, which is a wrapper around [tippy.js](https://atomiks.github.io/tippyjs).
- * @example
- * <Tooltip content='tooltip text' placement='top'><div>Hover here</div></Tooltip>
- * @example
- * const buttonRef = React.useRef();
- * ...
- * <Button ref={buttonRef} />
- * <Tooltip content='tooltip text' reference={buttonRef} />
- */
-export const Tooltip = (props: TooltipProps) => {
-  const { content, children, className, style, visible, ref, id, ...rest } =
-    props;
+const useTooltip = ({
+  placement = 'top',
+  visible: controlledOpen,
+  toggleVisible: setControlledOpen,
+  followTrigger = false,
+}: TooltipOptions = {}) => {
+  const [uncontrolledOpen, setUncontrolledOpen] = React.useState(false);
+
+  const open = controlledOpen ?? uncontrolledOpen;
+  const setOpen = setControlledOpen ?? setUncontrolledOpen;
+
+  const data = useFloating({
+    placement,
+    open,
+    onOpenChange: setOpen,
+    whileElementsMounted: (referenceEl, floatingEl, update) =>
+      autoUpdate(referenceEl, floatingEl, update, {
+        animationFrame: followTrigger,
+      }),
+    middleware: [offset(5), flip(), shift()],
+  });
+
+  const context = data.context;
+
+  const hover = useHover(context);
```

Comment:
Should be allow users to customise interactions? eg. disable tooltip on focus.

Reply:
it would be nice to give control, but i fear that developers will start removing focus as trigger, making tooltips inaccessible

Reply:
Not giving control for now. We can add that later.